Documentation/project-docs/benchmarking.md

```diff
@@ -0,0 +1,102 @@
+
+```
+public class MainConfig : ManualConfig
+{
+    public MainConfig()
+    {
+        Add(Job.Default
+            .With(Runtime.Core)
+            .With(CsProjCoreToolchain.From(new NetCoreAppSettings(
+                targetFrameworkMoniker: "netcoreapp2.1",
+                runtimeFrameworkVersion: "2.1.0-preview1-25919-02", // <-- Adjust version here
+                customDotNetCliPath: @"C:\Program Files\dotnet-nightly\dotnet.exe", // <-- Adjust path here
+                name: "Core 2.1.0-preview")))
+            .WithLaunchCount(1)
+            .WithWarmupCount(1)
+            .WithInvocationCount(10)
+            .WithUnrollFactor(1)
+            .WithTargetCount(3));
+
+        // Add whatever jobs you need
+        Add(DefaultColumnProviders.Instance);
+        Add(MarkdownExporter.GitHub);
+        Add(new ConsoleLogger());
+        Add(new HtmlExporter());
+        Add(MemoryDiagnoser.Default);
+    }
+}
+```
+
+
+
+### Using your self-compiled shared framework
+Follow the instructions described here https://github.com/dotnet/corefx/blob/master/Documentation/project-docs/dogfooding.md#more-advanced-scenario---using-your-local-corefx-build and skip the last part which calls the `dotnet.exe`.
+Make sure to build your local corefx repository in RELEASE mode `.\build -release`! You currently need to have a self-contained application to inject your local shared framework package.
+
+Now we are going to create a BenchmarkDotNet configuration file. Currently there is no easy way to run your BenchmarkDotNet application in a dedicated process, therefore we are using the InProcess switch:
+
+```
+public class MainConfig : ManualConfig
+{
+    public MainConfig()
+    {
+        Add(Job.Default
+            .With(InProcessToolchain.Instance) <-- To run against your local built shared framework (in corefx folder)
+            .WithLaunchCount(1)
+            .WithWarmupCount(1)
+            .WithInvocationCount(1)
+            .WithUnrollFactor(1)
+            .WithTargetCount(3));
+
+        // Add whatever jobs you need
+        Add(DefaultColumnProviders.Instance);
+        Add(MarkdownExporter.GitHub);
+        Add(new ConsoleLogger());
+        Add(new HtmlExporter());
+        Add(MemoryDiagnoser.Default);
+    }
+}
+```
+
+## Running the benchmark
+
+In your application entry point pass the configuration to the BenchmarkRunner:
+`BenchmarkRunner.Run<T>(new MainConfig()); // <-- Configuration class`
+
+As mentioned before, instead of calling the dotnet in the `PATH` we call the `dotnet.exe` explicitely from the downloaded SDK folder.
+To get valid results make sure to compile and run your project with RELEASE configuration:
+
+```
+> cd "path/to/your/benchmark/project"
+> "C:\Program Files\dotnet-nightly\dotnet.exe" restore
+> "C:\Program Files\dotnet-nightly\dotnet.exe" build -c Release
+> "C:\Program Files\dotnet-nightly\dotnet.exe" run -c Release
+```
```

Comment:
## Reporting results

Often in a PR or issue you will want to share performance results to justify a change. Benchmark.NET will have created a ???.md file which you can paste in, along with the code you benchmarked.